src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -78,6 +81,70 @@ public Storage storageForBucket(GoogleBucketResource bucketResource) {
     return gcsProjectFactory.getStorage(bucketResource.projectIdForBucket());
   }
 
+  /**
+   * Get all of the lines from any files matching the path, including wildcarded paths
+   *
+   * @param path path to files, or path including wildcard referring to many files
+   * @param projectId Project ID to use for storage service in case of requester pays bucket
+   * @return All of the lines from all of the files matching the path
+   */
+  public List<String> getGcsFilesLines(String path, String projectId) {
+    Storage storage = gcsProjectFactory.getStorage(projectId);
+    int lastWildcard = path.lastIndexOf("*");
+    String prefixPath = lastWildcard >= 0 ? path.substring(0, lastWildcard) : path;
+    return listGcsFiles(prefixPath, projectId, storage)
+        .flatMap(blob -> getGcsFileLines(blob, projectId, storage).stream())
+        .collect(Collectors.toList());
+  }
+
+  private Stream<Blob> listGcsFiles(String path, String projectId, Storage storage) {
+    GcsLocator locator = GcsPdao.getGcsLocatorFromGsPath(path);
+    Iterable<Blob> blobs =
+        storage
+            .list(
+                locator.getBucket(),
+                Storage.BlobListOption.prefix(locator.getPath()),
+                Storage.BlobListOption.userProject(projectId))
+            .iterateAll();
+    return StreamSupport.stream(blobs.spliterator(), false);
+  }
+
+  private List<String> getGcsFileLines(Blob blob, String projectId, Storage storage) {
+    String gsPath = GcsPdao.getGsPathFromBlob(blob);
+    logger.info("Getting lines from {}", gsPath);
+    String blobContents = GcsIO.getBlobContents(storage, projectId, blob);
+    return Arrays.stream(blobContents.split("\n")).collect(Collectors.toList());
```

Comment:
You can make a list directly from an array using `Arrays.asList()`. `List.of()` works here too.
```suggestion
    return Arrays.asList(blobContents.split("\n"));
```

```suggestion
    return List.of(blobContents.split("\n"));
```